substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AlignedHeapChunk.java

```diff
@@ -171,4 +182,34 @@ public UnsignedWord getAllocationStart(AlignedHeapChunk.AlignedHeader heapChunk)
             return getObjectsStart(heapChunk);
         }
     }
+
+    /**
+     * Supply a closure to be applied to {@link AlignedHeapChunk}s.
+     */
+    public interface Visitor {
```

Comment:
We also have `com.oracle.svm.core.MemoryWalker.Visitor` that we could use or extract `visitHeapChunk` into a new interface, but I think that code is subject to change.

Reply:
Actually, given that you visit chunks "raw" elsewhere and it's not particularly intricate, you could probably do it without this visitor.